packages/headless/src/controllers/commerce/context/headless-context.ts

```diff
@@ -0,0 +1,212 @@
+import { Schema } from "@coveo/bueno";
+import {requiredNonEmptyString} from '../../../utils/validate-payload';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {buildController, Controller} from '../../controller/headless-controller';
+
+const optionsSchema = new Schema({
+  url: requiredNonEmptyString,
```

Comment:
Granted that we remove the `cart`, `view`, and `labels` attribute from this controller's state, I think we should expose all of the remaining attributes in its options schema.

Here my reasoning:

- Once set, `trackingId` is likely to remain static during the entire browsing session. In fact, can there realistically even be multiple tracking IDs within the same commerce site? I suppose if it's theoretically possible, then it's OK to expose an option to mutate that attribute, but otherwise we should only allow it to be set when initializing the controller.

- The `clientId` and the `user` context are kind of similar. In most sessions, those attributes shouldn't change at all after initialization, unless the user logs out and back in as another user or something. Exposing methods to mutate those attributes is still legitimate, though.

- `language` and `currency` are perhaps more likely to change, but still in most sessions, they'll probably be entirely static. Again, exposing methods to mutate those is legit.

In conclusion, I see the global commerce context as a thing that would typically be set only once in the context of a browsing session and not really mutated thereafter, so I think it makes sense to allow all of its initial state to be specified through its config options.

Reply:
Agreed! Left them exposed on this controller.